app-modules/engagement/src/Models/EngagementBatch.php

```diff
@@ -130,4 +131,15 @@ public static function renderWithMergeTags(string $html): string
 
         return $dom->saveHTML();
     }
+
+    /**
+     * @todo Remove this dynamic cast once `RefactorEngagementCampaignSubjectToJsonb` is removed.
+     *       Move 'subject' casting to the static `$casts` array: protected $casts = ['subject' => 'array'];
+     */
+    protected function casts(): array
```

Comment:
Does this overwrite the `$casts` property? If so, we will need to make sure that the other casts are in here as well.